src/transformers/tokenization_utils.py

```diff
@@ -346,13 +346,18 @@ class PreTrainedTokenizer(PreTrainedTokenizerBase):
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
 
-        # Added tokens - We store this for both slow and fast tokenizers
-        # until the serialization of Fast tokenizers is updated
-        self.added_tokens_encoder: Dict[str, int] = {}
-        self.added_tokens_decoder: Dict[int, str] = {}
-        self.unique_no_split_tokens: List[str] = []
-        self.tokens_trie = Trie()
+        if self.unk_token is not None and self.unk_token not in self._added_tokens_decoder:
```

Comment:
This avoids any potential errors. If there are no unk token then we are gonna be stuck in loop.